sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -338,159 +348,293 @@ export class StreamingReceiver extends MessageReceiver {
+
+    if (messageHandlers.forwardInternalErrors) {
+      const errorArgs: InternalProcessErrorArgs = {
+        error,
+        entityPath: this.entityPath,
+        errorSource: "internal",
+        fullyQualifiedNamespace: this._context.config.host
+      };
+
+      return messageHandlers.processError(errorArgs as ProcessErrorArgs);
+    }
+
+    return Promise.resolve();
+  }
+
   private _getHandlers(): ReceiverHandlers {
     return {
       onMessage: (context: EventContext) =>
-        this._onAmqpMessage(context).catch(() => {
-          /* */
-        }),
+        this._onAmqpMessage(context).catch((err) => this._reportInternalError(err)),
       onClose: (context: EventContext) =>
-        this._onAmqpClose(context).catch(() => {
-          /* */
-        }),
+        this._onAmqpClose(context).catch((err) => this._reportInternalError(err)),
       onSessionClose: (context: EventContext) =>
-        this._onSessionClose(context).catch(() => {
-          /* */
-        }),
+        this._onSessionClose(context).catch((err) => this._reportInternalError(err)),
       onError: this._onAmqpError,
       onSessionError: this._onSessionError
     };
   }
 
   async stopReceivingMessages(): Promise<void> {
     await this._receiverHelper.suspend();
+
+    if (this._subscribeCallPromise) {
+      await this._subscribeCallPromise;
+    }
+  }
+
+  async close(): Promise<void> {
+    await this._receiverHelper.suspend();
+    return super.close();
   }
 
+  private _subscribeCallPromise: Promise<void> | undefined;
+
   /**
-   * Initializes the link. This method will retry infinitely until a connection is established.
+   * Starts the receiver by establishing an AMQP session and an AMQP receiver link on the session.
    *
-   * The retries are broken up into cycles. For each cycle we do a set of retries, using the user's
-   * configured retryOptions. If that retry call fails we will report the error and then go into a
-   * new cycle, repeating the retries the same as before.
+   * Any errors thrown by this function will also be sent to the messageHandlers.processError function
+   * _and_ thrown, ultimately from this method.
    *
-   * It is completely up to the user to break out of this retry cycle in their error handler by either:
+   * NOTE: This function retries _infinitely_ until success! It is completely up to the user to break
+   * out of this retry cycle otherwise by:
    * 1. closing the receiver
    * 2. Calling `close` on the subscription instance they received when they initially called subscribe().
-   * 3. aborting the abortSignal they passed in when calling subscribe (this does not apply in onDetached, however)
+   * 3. aborting the abortSignal they passed in when calling subscribe (this also applies to initialization calls in onDetach)
+   *
+   * @param onMessage - The message handler to receive servicebus messages.
+   * @param onError - The error handler to receive an error that occurs while receivin messages.
```

Comment:
These parameters do not apply to the below method...

Reply:
Ah, good spot. I always forget the doc comments!